### PR TITLE
Add Eio benchmark

### PR DESCRIPTION
**Pull request by talex5**
This added a benchmark using [eio](https://github.com/ocaml-multicore/eio). Eio is a work-in-progress and is slightly slower than the existing `httpaf-effects` benchmarks at present on my machine.

### CODE REVIEW

**Comment by talex5** [comment]
BTW, I'm testing it with: `docker run --ulimit memlock=819200000:819200000 --security-opt seccomp=unconfined` on Linux 5.12.

**Comment by sadiqj** [comment]
Any idea why there's a missing angstrom for the build now?

**Comment by talex5** [comment]
Maybe the CI doesn't checkout submodules?

File: httpaf-eio/wrk_effects_benchmark.ml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+let text = Bigstringaf.of_string ~off:0 ~len:(String.length text) text
+
+let headers = Headers.of_list ["content-length", string_of_int (Bigstringaf.length text)]
+let request_handler _ reqd =
+  let request = Reqd.request reqd in
+  match request.target with
+  | "/" ->
+    let response_ok = Response.create ~headers `OK in
```
Comment (avsm) [inline httpaf-eio/wrk_effects_benchmark.ml:13]: We're pre-allocating the `headers` variable, but then dynamically allocating the `response_ok`. Could probably just make the whole response static?

Comment (talex5) [reply httpaf-eio/wrk_effects_benchmark.ml:13]: This is just the same as the other benchmarks:

https://github.com/ocaml-multicore/retro-httpaf-bench/blob/80b118f020ae8b5352fe2b41098f7c17da3b1f35/httpaf-effects/wrk_effects_benchmark.ml#L26

Comment (sadiqj) [reply httpaf-eio/wrk_effects_benchmark.ml:13]: I removed the preallocation from the rust benchmark to match our other benchmarks when I initially added it.

**Comment by talex5** [comment]
I updated this PR to the latest version of Eio. CI is passing now.

**Comment by sadiqj** [comment]
Thanks!

**Pull request merged**